test/e2e/framework/util.go

```diff
@@ -506,6 +506,17 @@ func LookForStringInPodExec(ns, podName string, command []string, expectedString
 	})
 }
 
+// LookForStringInPodExecToContainer looks for the given string in the output of a command
+// executed in a specific pod container.
+func LookForStringInPodExecToContainer(ns, podName, containerName string, command []string, expectedString string, timeout time.Duration) (result string, err error) {
```

Comment:
another nit: this kind of code seems generally useful.
I think this would be better in https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework/pod.
`/test/e2e/framework/util.go` should be for functions directly related to the e2e test framework.

Reply:
Thanks for the review! Actually, the LookForStringInPodExec function [above](https://github.com/kubernetes/kubernetes/pull/92127/files/426d0bbcb13489c50a52dbbf3b5fef24bb2720d6#diff-eb7b79470992813ea1905e96c298b47bL500) should have been the function I added. The comments on that function mention look for a given string in specific container, but the container name seems to have been missed. I added this so as to not disturb existing references to the above function as part of this very large PR if I fixed the above function.

Reply:
aaah I see. And I also just noticed the comment on the above function - the ideal place for these functions would be not in `e2e/framework/util.go` (for the sake of code organization) but moving functions that use `lookFrString` is non-trivial because they depend on a lot of other code.

Thank you for your reply!

Reply:
Actually, there are just 17 references to LookForStringIn[PodExec|Log] functions.

It has been bothering me that I need to add ```LookForStringInPodExecToContainer``` where it should really have been LookForStringInPodExec that takes container name as parameter, and defaults to first container if name is empty. There are just 9 references to LookForStringInPodExec at this time, so it is a very manageable churn.

I'm happy to send a refactoring PR for the above, and get rid of the need to add LookForStringInPodExecToContainer - I just don't want it to be part of this large PR. 

Please let me know if you'd like me to create an issue and refactor this - I can certainly move them to another more suitable location while I'm at it. The refactor has to go in first. We are now targeting this PR for v1.20, so there is not much rush from me to get this in.